api/management/commands/ingest_ns_initiatives.py

```diff
@@ -24,9 +24,9 @@ def handle(self, *args, **kwargs):
         cbf_url = requests.get(f"https://data-api.ifrc.org/api/cbf?apikey={api_key}")
 
         # resposne for individual request
-        esf_response = esf_url.json()
-        nsia_response = nsia_url.json()
-        cbf_response = cbf_url.json()
+        esf_response = esf_url.json() if esf_url.status_code == 200 else []
+        nsia_response = nsia_url.json() if nsia_url.status_code == 200 else []
+        cbf_response = cbf_url.json() if cbf_url.status_code == 200 else []
```

Comment:
```suggestion
   urls = [
            f"https://data-api.ifrc.org/api/esf?apikey={api_key}",
            f"https://data-api.ifrc.org/api/nsia?apikey={api_key}",
            f"https://data-api.ifrc.org/api/cbf?apikey={api_key}",
        ]

        responses = []
        for url in urls:
            response = requests.get(url)
            responses.append(response.json()) if response.status_code == 200 else responses.append([])

        added = 0
        flatList = [element for innerList in responses for element in innerList]
```

Reply:
I think no "append nothing" is needed if the status_code is not 200.
(In the above "if"-s it was mandatory to give an "else" option.)

Reply:
Thanks, I used your suggestions, the code became cleaner.